packages/store/src/StoreClass.tsx

```diff
@@ -253,6 +255,63 @@ export class StoreClass<T extends MachinesObj> implements IStore<T> {
     const service = this.services[key];
     return waitFor(service, ...args);
   }
+
+  /**
+   * Wait for a specifice state to be reached.
+   * @param key - the key of the service
+   * @param done - the state to wait for complete (default: 'done')
+   * @param failure - the state to wait for failure (default: 'failed')
+   * @param failureMessage - the field of the context to use as the error message (default: 'error')
+   * @param timeout - the timeout in milliseconds (default: 5000)
+   * @returns the current state value
+   * @example
+   * await store.waitForState('counter');
+   * // or
+   * await store.waitForState('counter', 'done', 'failed', 'error');
+   */
+  public async waitForState<K extends keyof T, S extends StateItem<T, K>>(
+    ...[
+      key,
+      {
+        done = 'done',
+        failure = 'failed',
+        failureMessage = 'error',
+        timeout = 5000,
+      } = {},
+    ]: WaitForStateArgs<T, K>
+  ) {
+    try {
+      const service = this.services[key];
+      const doneState = done || 'done';
+      const failureState = failure || 'failed';
+      const failureMessageField = failureMessage || 'error';
+
+      if (!service) {
+        throw new Error('Service not found');
+      }
+
+      const appState = await waitFor<AnyInterpreter>(
+        service,
+        (state) => state.matches(doneState) || state.matches(failureState),
+        timeout
+      );
+
+      if (appState.matches(failureState)) {
+        throw new Error(appState.context[failureMessageField], {
+          cause: 'CustomState',
+        });
+      }
+
+      return appState as S;
+    } catch (err: unknown) {
+      if (err && (err as { cause: string }).cause === 'CustomState') {
+        throw err as Error;
+      }
+      throw new Error(
```

Comment:
Is there a way to tell that the inactivity error is actually the error we are encountering?  I'm just worried that this will throw the inactivity error when it is actually something else failing.

Reply:
This is a good idea, I will take a look at doing this.

Reply:
if it's something else that we know, would be detected in first conditional, right @luizstacio ??

Reply:
Not really as the first one only checks for expected states of error I add a new implementation to check if the error is timeout if is not will throw the err.